client/src/components/Lesson/LessonCreator.tsx

```diff
@@ -60,14 +58,6 @@ export class LessonCreator extends BindingComponent<LessonCreatorProps, Lesson>
         return this.state;
     }
 
-    componentWillMount() {
-        if (this.props.notFound) {
```

Comment:
Ah, since we're not driving lesson loading through the component, we can probably infer this when value is undefined